### DISCUSSION TITLE
Multi-tenancy support

### DISCUSSION THREAD

**Discussion by deepakprabhakara** [vectordotdev/vector#18691] [General]
Many thanks for a great project!

What's the best way to allow dynamic sources and sinks to be defined? We would like to avoid having to orchestrate multiple instances of Vector if we can. I believe there are no APIs to manage sources and sinks on a running Vector instance. What are the possible ways to go about this?

**Comment by jszwedko** [comment]
Hi @deepakprabhakara !

The idea of having an API to manage components has been discussed but unfortunately I think the easiest way to do that at the moment would be to have an external process that acts as the "API", writes configuration updates to a file, and uses Vector's reload mechanism (issuing it a `SIGHUP`) to pick up the changes.

**Reply by deepakprabhakara** [reply]
Thanks @jszwedko, that might work well. Just to clarify, SIGHUP only reload configuration and rest of Vector continues to service existing requests?

**Comment by deepakprabhakara** [comment]
SIGHUP works well and reloads config, many thanks @jszwedko
